src/backend/libc/net/netdevice.rs

```diff
@@ -0,0 +1,56 @@
+#![allow(unsafe_code)]
+
+#[cfg(feature = "alloc")]
+use crate::alloc::string::String;
+use crate::backend::io::syscalls::ioctl;
+use crate::fd::AsFd;
+use crate::io;
+use crate::net::netdevice::open_socket;
+#[cfg(feature = "alloc")]
+use libc::SIOCGIFNAME;
+use libc::{__c_anonymous_ifr_ifru, c_char, ifreq, IFNAMSIZ, SIOCGIFINDEX};
+
+#[cfg(target_os = "linux")]
+pub(crate) fn name_to_index(if_name: &str) -> io::Result<u32> {
+    let if_name_bytes = if_name.as_bytes();
+    if if_name_bytes.len() >= IFNAMSIZ as usize {
+        return Err(io::Errno::NODEV);
+    }
+
+    let mut ifreq = ifreq {
+        ifr_name: [0; 16],
+        ifr_ifru: __c_anonymous_ifr_ifru { ifru_ifindex: 0 },
+    };
+
+    let mut if_name_c_char_iter = if_name_bytes.iter().map(|byte| *byte as c_char);
+    ifreq.ifr_name[..if_name_bytes.len()].fill_with(|| if_name_c_char_iter.next().unwrap());
+
+    let fd = open_socket()?;
+    unsafe { ioctl(fd.as_fd(), SIOCGIFINDEX as _, &mut ifreq as *mut ifreq as _) }?;
+    let index = unsafe { ifreq.ifr_ifru.ifru_ifindex };
+    Ok(index as u32)
+}
+
+#[cfg(all(target_os = "linux", feature = "alloc"))]
+pub(crate) fn index_to_name(index: u32) -> io::Result<String> {
+    let mut ifreq = ifreq {
+        ifr_name: [0; 16],
+        ifr_ifru: __c_anonymous_ifr_ifru {
+            ifru_ifindex: index as _,
+        },
+    };
+
+    let fd = open_socket()?;
```

Comment:
It's unusual from rustix's perspective that this is creating a new socket, rather than having the user pass in a socket. It's an extra system call, which may fail, or which users may want more control over. Would it be ok to change the public API here to have an `fd` parameter, with documentation telling users how to create the `fd` if needed, rather than having this `open_socket` function call?

Reply:
I changed the API accordingly and added docs what glibc and Android do and what the manual specifies.